types/persisted-model.d.ts

```diff
@@ -519,6 +520,10 @@ export declare class PersistedModel extends ModelBase {
    * @returns {String} The `id` property name
    */
   static getIdName(): string;
+
+  definePersistedModel<T>(entityClass: T): PersistedModel;
+
+  static observe(operation: string, handler: Listener): void;
```

Comment:
While you are adding `observe` method, can you please add typings for other methods related to operation hooks?

**SHOULD HAVE**

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/observer.js#L76

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/observer.js#L99

**NICE TO HAVE**

I think these APIs are mostly internal. Do they support promise-based invocations?

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/observer.js#L131

https://github.com/strongloop/loopback-datasource-juggler/blob/36303b3b63932478cb429b5c5af6fec7f09b602c/lib/observer.js#L221

IIRC, we often copy API documentation from jsdocs into tsdoc comments, so that consumers of the TypeScript typedefs can get API docs rendered in their IDE like VSCode.

Reply:
`notifyObserversOf ` and `notifyObserversAround ` are for internal uses, let's omit them.